### PR TITLE
Manual platforms

### PR DESCRIPTION
**Pull request by jmacdotorg**
This addresses #201 .

* Adds a 'platform' field to the entry table - it's an `enum` with a list of possible values matching what the current website recognizes

* Removes all platform-detection code

* Since `inform` or `glulx` games are no longer re-classified as `parchment` or `quixe` as soon as they're processed (because the web app turns them into websites, and thus the auto-detection code would see them differently as soon as that happens), we add these platforms to the list of games that get "Play" buttons on the ballot

* Modifying how the URL of play files (if any) is determined, and adding tests for it.

### CODE REVIEW

File: IFComp/lib/IFComp/Form/Entry.pm [hunk 1]
```diff
@@ -27,6 +27,66 @@ has_field 'subtitle' => (
...
+        {
+            value => 'quest',
+            label => 'Quest (download-only)',
+        },
+        {
+            value => 'quest-website',
+            label => 'Quest (online play available)',
+        },
+        {
+            value => 'tads',
```
Comment (curiousdannii) [inline IFComp/lib/IFComp/Form/Entry.pm:64]: I suggest you split this up into TADS 3 and TADS 3 Web UI, as handling them will probably be different.

File: IFComp/lib/IFComp/Form/Entry.pm [hunk 2]
```diff
@@ -27,6 +27,66 @@ has_field 'subtitle' => (
...
+        {
+            value => 'quixe',
+            label => q{Glulx, using Inform's "Release with a website" feature},
+        },
+        {
+            value => 'parchment',
+            label => q{Z-code, using Inform's "Release with a website" feature},
+        },
+        {
+            value => 'inform-website',
```
Comment (curiousdannii) [inline IFComp/lib/IFComp/Form/Entry.pm:52]: What's the value of having these three Inform website options here? How will they be handled differently?

File: IFComp/lib/IFComp/Form/Entry.pm (hunk 1 above)
Comment (jmacdotorg) [reply IFComp/lib/IFComp/Form/Entry.pm:64]: This is a good idea, but my goal here is to keep the list of possible platforms exactly as it was before this change. We could introduce new possible platforms (and rules about handling them) as a future step.

File: IFComp/lib/IFComp/Form/Entry.pm (hunk 2 above)
Comment (jmacdotorg) [reply IFComp/lib/IFComp/Form/Entry.pm:52]: It's mostly for labeling purposes on the ballot.

File: IFComp/lib/IFComp/Form/Entry.pm (hunk 1 above)
Comment (curiousdannii) [reply IFComp/lib/IFComp/Form/Entry.pm:64]: They are substantially different platforms, and people are discussing them on the forum now. I'm not sure if Web UI games have ever been submitted before, but they really shouldn't be lumped together. So don't forget to do the follow up step ;)

**Comment by adherzog** [comment]
It looks like `IFComp::Blorb` is no longer being used. You should probably remove it (and the reference in `IFComp::Schema::Result::Entry`.)

**Comment by jmacdotorg** [comment]
This is parked while going through some iterative testing by QA. (See #217.)

**Pull request merged**